index/src/transactors/evm_transactor.rs

```diff
@@ -0,0 +1,96 @@
+use crate::traits::Error;
+use alloc::string::String;
+use pink_web3::api::{Eth, Namespace};
+use pink_web3::contract::tokens::Tokenize;
+use pink_web3::contract::{Contract, Options};
+use pink_web3::ethabi::{Bytes, Token, Uint};
+use pink_web3::keys::pink::KeyPair;
+use pink_web3::signing::Key;
+use pink_web3::transports::{resolve_ready, PinkHttp};
+use pink_web3::types::{Res, H160, H256};
+
+/// The client to submit transaction to the Evm evm_contract contract
+pub struct EvmContractClient {
+    pub contract: Contract<PinkHttp>,
+}
+
+impl EvmContractClient {
+    /// Calls the EVM contract `deposit` function
+    ///
+    /// # Arguments
+    ///
+    /// * `dest_chain_id` - ID of chain deposit originated from.
+    /// * `token_rid` - resource id used to find address of token handler to be used for deposit
+    /// * `data` - Addition data to be passed to special handler
+    pub fn deposit(
+        &self,
+        signer: KeyPair,
+        token_rid: H256,
+        amount: Uint,
+        recipient_address: Bytes,
+    ) -> core::result::Result<H256, Error> {
+        let data = Self::compose_deposite_data(amount, recipient_address);
```

Comment:
Note ChainId 1 only represents Khala, other ChainId are:
- Ethereum: 0
- Moonriver: 1
- Phala: 3
Hard code is ok for demonstration

Reply:
yes, will fix it

Reply:
Moonriver should be 2, sorry for the mistake

Reply:
fixed